File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/nodes/parameters/NodesFilters.java
package co.uk.swarmbit.docker.api.nodes.parameters;

import co.uk.swarmbit.docker.api.common.AbstractFilters;
import co.uk.swarmbit.docker.api.common.Filters;

public class NodesFilters extends AbstractFilters implements Filters {

    public NodesFilters setId(String id) {
        addFilter("id", id);
        return this;
    }

    public NodesFilters setLabel(String label) {
        addFilter("label", label);
        return this;
    }

    public NodesFilters setMembership(NodeMembership membership) {
        addFilter("membership", membership.toString());
        return this;
    }

    public NodesFilters setName(String name) {
        addFilter("name", name);
        return this;
    }

    public NodesFilters setRole(NodeRole role) {
        addFilter("role", role.toString());
        return this;
    }

}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/nodes/parameters/NodesFilters.kt
package com.swarmbit.docker.api.nodes.parameters

import com.swarmbit.docker.api.common.AbstractFilters
import com.swarmbit.docker.api.common.annotation.DockerRemoteApiMinVersion

class NodesFilters : AbstractFilters() {

    fun setId(id: String): NodesFilters {
        addFilter("id", id)
        return this
    }

    fun setLabel(label: String): NodesFilters {
        addFilter("label", label)
        return this
    }

    fun setMembership(membership: NodeMembership): NodesFilters {
        addFilter("membership", membership.toString().toLowerCase())
        return this
    }

    fun setName(name: String): NodesFilters {
        addFilter("name", name)
        return this
    }

    fun setRole(role: NodeRole): NodesFilters {
        addFilter("role", role.toString().toLowerCase())
        return this
    }

    @DockerRemoteApiMinVersion("v1.40")
    fun setNodeLabel(nodeLabel: String, equals: Boolean = true): NodesFilters {
        addFilter("node.label", nodeLabel, equals)
        return this
    }
}
